[{"count": 0, "tags": [], "creator": "chris@blaze.io", "is_private": false, "text": "Access logs written by org.apache.catalina.valves.AccessLogValve are not being written for servlet 3.0 async requests.\n\nTo reproduce:\n- Handle a request asynchronously by marking the servlet with @WebServlet(asyncSupported=true)\n- Send some traffic and notice that nothing is being written to the access logs.\n\nAccessLogValve.log() is called by org.apache.catalina.Context.logAccess().\nFrom the source for version 7.0.11, in org.apache.catalina.connector.CoyoteAdapter.java, line 416:\n\n            AsyncContextImpl asyncConImpl = (AsyncContextImpl)request.getAsyncContext();\n            if (asyncConImpl != null) {\n                async = true;\n            } else if (!comet) {\n                response.finishResponse();\n                if (postParseSuccess) {\n                    // Log only if processing was invoked.\n                    // If postParseRequest() failed, it has already logged it.\n                    ((Context) request.getMappingData().context).logAccess(\n                            request, response,\n                            System.currentTimeMillis() - req.getStartTime(),\n                            false);\n                }\n                req.action(ActionCode.POST_REQUEST , null);\n            }\n\nIn the async case logAccess() isn't called here. I've searched through the code and don't see logAccess() being called anywhere else, except in error cases. So, it looks like the async case is missing a call to logAccess() somewhere.", "id": 145620, "time": "2011-04-07T11:18:30Z", "bug_id": 51038, "creation_time": "2011-04-07T11:18:30Z", "attachment_id": null}, {"count": 1, "tags": [], "creator": "markt@apache.org", "is_private": false, "text": "This has been fixed in trunk and will be included in 7.0.13 onwards.", "id": 145821, "time": "2011-04-16T17:24:09Z", "bug_id": 51038, "creation_time": "2011-04-16T17:24:09Z", "attachment_id": null}]